### ISSUE TITLE
Using "back to.." starts replay instead of allowing to continue to play as before

### ISSUE THREAD

**Issue by knyghtmare** [wesnoth/wesnoth#7491]
### Game and System Information

- **Version:** 1.17.14+dev
- **Downloaded from:**  Github
- **Build info:** Msys2
- **OS:** Windows 10

### Description of the bug

Using "back to.." after scenario Defeat permanently converts scenario to replay save and goes on loop...

### Steps to reproduce the behavior

1. Download Save
2. Load Save
3. Try to survive the Spectre and 2 ghosts
4. Get Malin killed
5. Back to previous turn (back to turn 17),
6. Endure infinite loop of Malin dying over and over

### Expected behavior

I can play the previous turn without it being a replay

### Additional context

[DiD-Descent into Darkness-Auto-Save17.gz](https://github.com/wesnoth/wesnoth/files/11080311/DiD-Descent.into.Darkness-Auto-Save17.gz)
(might need to install the Unitmarker as i was using it)

**Comment by gfgtdf** [comment]
> I can for turn 16 but the turn 17, Malin attacks, kills one, and then gets thrashed by the Spectre on loop, I cannot press "Continue Play" for turn 17

Soo you are not quick enough ?:p

> Well, duh. But I really should not be seeing the replay UI when "Back to " for an ongoing playthrough...

I disagree here, For the reasons mentioned in [https://github.com/wesnoth/wesnoth/issues/7460 and https://github.com/wesnoth/wesnoth/commit/790948f623b36ad29ecbec1ccc915417752658e5](https://github.com/wesnoth/wesnoth/commit/790948f623b36ad29ecbec1ccc915417752658e5) i think this can be quite useful, (and the simple loading save behaviour is still available via load game menu)

**Comment by cooljeanius** [comment]
I agree with @knyghtmare; I wouldn't expect to see replay UI when using "Back to" in an existing playthrough.

**Comment by soliton-** [comment]
So there is no loop and you can "reset to beginning" and "Continue Play" from the replay controls to get the previous behaviour. I don't see why this change in behaviour is desirable though. The replay feature should be under a different menu entry and going back to a previous turn should continue to work as before.

**Comment by knyghtmare** [comment]
> The replay feature should be under a different menu entry and going back to a previous turn should continue to work as before.

Yes, please restore it. The new "feature" is frustrating IMHO.

**Comment by gfgtdf** [comment]
For now i changed it so that  the replay doesn't automaticially start anymore and people can press "continue play" immidiately, adding a second menu is a bit more complicated as the menu code is rather complex, so it will take some time.

**Comment by ForestDragon-wesnoth** [comment]
> > I don't see why this change in behaviour is desirable though.
> 
> I still think it is, because it allows player to easily load not only from the beginning of their turn, but from any point during their turn. (Or during the enemies turn, but that probably doesn't matter since the ai will immidiately take over control). For a player who likes to save a lot, this means that he no longer has to worry about saving at different times, he can just go back to the turn where wants to reload from and then skip until he has reached the position he wants to load from and press "continue play". And the only downside is one more button press "Continue game" when loading an autosave.

That could potentially be useful, yes, but I firmly believe that such a feature should be optional instead of overriding default behavior. The potential benefits are currently far outweighed by the annoyance and confusing the UI change causes

**Comment by knyghtmare** [comment]
So, using "Back to turn X" in SP now takes you back to the Last AI side's turn X ...not mine (player/ side= 1)...I think this is a bug now...

**Comment by gfgtdf** [comment]
> So, using "Back to turn X" in SP now takes you back to the Last AI side's turn X ...not mine (player/ side= 1)...I think this is a bug now...

This is really strange, do you get the same behavior when you load the same auosave file via the load game dialog?

> But it's not about the replay feature at all. That is very useful. There is just no reason to replace existing functionality.

Yes, i think a second menu could work for now, it just will take a while to implement

> I mean the feature was apparently already there for multiplayer and who even knew about it because it was behind the existing menu entry? I certainly only found out when you mentioned it a couple days ago.

Hmm we should probably promote these better, (but tbh. when i added the feature in initially it was a bit unstable so i didn't want to promote it too much, and its hard to figure out the exact timing on when to promote these things)

Maybe we should add a "tip of the say" (i mean the titlescreen thing) for that?

**Comment by knyghtmare** [comment]
> This is really strange, do you get the same behavior when you load the same auosave file via the load game dialog?

Happens with "back to", context is player (some of them) would revert to the previous turn as soon as one of their favored units died, or heroes/leaders so this happens during enemy turn so this led to the loaded game showing the previous turn of the last enemy side. Load game one does not

**Comment by gfgtdf** [comment]
hmm did you test with current master, in particular with https://github.com/wesnoth/wesnoth/commit/549b0ae31fd68f15c2512e37cd645406c2331ed2 ?

**Comment by gfgtdf** [comment]
So you are saying that 
1) When you use "Back To" and then "continue Play" you are not in the start of your turn but in the start of the enemies turn?
2) This does not happen when the corepsonding  state is loaded via the load game menu.

If yes then the only thing i could think of that could be happening is that  when the new "Back To" is used the state of the "End Turn" button is somehow not reset properly, and the players turn is ended immidiately. Will look into this.

**Comment by knyghtmare** [comment]
Yes and yes.

> So you are saying that
> 
> 1. When you use "Back To" and then "continue Play" you are not in the start of your turn but in the start of the enemies turn?
> 2. This does not happen when the corresponding  state is loaded via the load game menu.

Yes and yes

**Comment by gfgtdf** [comment]
> When you use "Back To" and then "continue Play" you are not in the start of your turn but in the start of the enemies turn?

I think this should be fixed by 
https://github.com/wesnoth/wesnoth/commit/8bcd731f7a6f4feb99bb6682fd50f0f7b8c93ff9

I wasn't able to reproduce it yet, so i'm not 100% sure though.

**Comment by gfgtdf** [comment]
There are now different menu items for for replays and loading an autosave

**Comment by Wedge009** [comment]
Is this still an issue, given it's been marked as ready for testing?
